rust/core/src/driver_manager.rs

```diff
@@ -316,6 +408,65 @@ impl Driver for ManagedDriver {
     }
 }
 
+fn get_optional_key(manifest: &Table, key: &str) -> String {
+    manifest
+        .get(key)
+        .and_then(Value::as_str)
+        .unwrap_or_default()
+        .to_string()
+}
+
+fn load_driver_manifest(info: &mut DriverInfo) -> Result<()> {
+    let contents = fs::read_to_string(info.manifest_file.as_path()).map_err(|e| {
+        let file = info.manifest_file.to_string_lossy();
+        Error::with_message_and_status(
+            format!("Could not read manifest '{file}': {e}"),
+            Status::InvalidArguments,
+        )
+    })?;
+
+    let manifest = contents
+        .parse::<Table>()
+        .map_err(|e| Error::with_message_and_status(e.to_string(), Status::InvalidArguments))?;
```

Comment:
I opened https://github.com/apache/arrow-adbc/pull/3101 with an alternative approach to handle `Manifest` deserialization.

Reply:
PLEASE let's not include `serde`. TOML was chosen as the format for manifests and that's the only ser/de format that the driver manager will need.

@paleolimbot 💯. Better error messages, less binary code bloat, better abstractions... etc.

Reply:
> Is that maybe overkill given that we aren't looking to provide full manifest interactions, just a simple parse to get the shared lib path?

By providing a `Manifest` definition (behind a feature flag) users can easily use the other fields, because even if we don't use those other fields here now, others might.

> How do I modify that to express that the `shared` member could be _either_ a string or a `map<string, string>` that maps platform triples to a path?

https://serde.rs/string-or-struct.html

> PLEASE let's not include `serde`. TOML was chosen as the format for manifests and that's the only ser/de format that the driver manager will need.

`serde` doesn't include other data formats (`Deserializer` implementations)?

> better abstractions

Can you elaborate?

Reply:
I've updated this to use the serde implementation that @mbrobbel suggested. While it does make the implementation of `load_driver_manifest` in `driver_manager.rs` much simpler, it is at the expense of the complexity of the `manifest.rs` file itself and custom serde etc.

I'll let all of you decide what makes the most sense here as I'm not as familiar with the benefits or drawbacks here. @phillipleblanc @lidavidm either of you have an opinion here also?

Reply:
In general, I favor libraries being thoughtful about which dependencies they require to work. I'd also lean towards dropping serde to keep the original parsing logic.

Reply:
given we have three people in favor of removing serde and using the original logic, I'm gonna switch back to that unless someone objects.